### PR TITLE
Add support for sql_definition parameter to update_view

### PR DESCRIPTION
**Pull request by jwood**
This pull requests adds support for the `sql_definition` parameter when updating views.

### CODE REVIEW

File: lib/scenic/statements.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ def create_view(name, version: 1, sql_definition: nil, materialized: false)
       if version.blank? && sql_definition.nil?
         raise(
           ArgumentError,
-          "view_definition or version_number must be specified"
+          "sql_definition or version must be specified",
```
Comment (calebthompson) [inline lib/scenic/statements.rb]: The argument still appears to be version_number here.

File: lib/scenic/statements.rb [hunk 2]
```diff
@@ -66,6 +66,9 @@ def drop_view(name, revert_to_version: nil, materialized: false)
     #
     # @param name [String, Symbol] The name of the database view.
     # @param version [Fixnum] The version number of the view.
+    # @param sql_definition [String] The SQL query for the view schema. If both
+    #   `sql_defintiion` and `version` are provided, `sql_definition` takes
```
Comment (calebthompson) [inline lib/scenic/statements.rb]: Typo defintiion

File: lib/scenic/statements.rb (hunk 1 above)
Comment (calebthompson) [inline lib/scenic/statements.rb]: Never mind I need more coffee

**Comment by calebthompson** [comment]
@jwood just a typo fix, then please squash your commits.

**Comment by calebthompson** [comment]
Failing test seems to be a Travis problem.

**Comment by jwood** [comment]
@calebthompson Typo fixed, and commits squashed.

**Comment by derekprior** [comment]
What's the use case here?

Also, I wouldn't necessarily worry about the breaking change. As far as I understand it, it's a bug we are fixing.

**Comment by calebthompson** [comment]
@derekprior my understanding is that this will bring us into parity with create_view's sql_definition option, which makes sense to me.

**Comment by jwood** [comment]
The use case is that we need to update a view that we used `create_view` to create, using `create_view`'s sql_definition option. We have a SQL ERB template that we first feed through ERB, and then feed the generated SQL to `create_view`/`update_view`.

**Comment by jwood** [comment]
Hi @calebthompson and @derekprior. Thanks for all the feedback. I'm just looking to get a little clarification on the change @calebthompson suggested here: https://github.com/thoughtbot/scenic/pull/162#commitcomment-18464891

I'm more than happy to make the change, but if we do, I think we should change both `create_view` and `update_view` for consistency. And, that would be a backwards incompatible change for `create_view` (not for `update_view`, because that would be new functionality).

Please let me know if you would like me to raise an error if both the `version` and the `sql_definition` options are provided.

**Comment by jwood** [comment]
Hi @calebthompson and @derekprior. Any thoughts on how to push this through?

**Comment by calebthompson** [comment]
> Please let me know if you would like me to raise an error if both the version and the sql_definition options are provided.

We would, and I think it's a bug that we didn't before and instead failed silently, so it's not backward-incompatible.

File: lib/scenic/statements.rb [hunk 3]
```diff
@@ -6,9 +6,9 @@ module Statements
     # @param name [String, Symbol] The name of the database view.
     # @param version [Fixnum] The version number of the view, used to find the
     #   definition file in `db/views`. This defaults to `1` if not provided.
-    # @param sql_definition [String] The SQL query for the view schema. If both
-    #   `sql_defintiion` and `version` are provided, `sql_definition` takes
-    #   prescedence.
+    # @param sql_definition [String] The SQL query for the view schema. An error
+    #   will be raised if `sql_defintiion` and `version` are both set,
```
Comment (calebthompson) [inline lib/scenic/statements.rb]: Do you mind fixing the typo here as well?

File: lib/scenic/statements.rb [hunk 4]
```diff
@@ -75,12 +78,15 @@ def drop_view(name, revert_to_version: nil, materialized: false)
     # @example
     #   update_view :engagement_reports, version: 3, revert_to_version: 2
     #
-    def update_view(name, version: nil, revert_to_version: nil, materialized: false)
-      if version.blank?
-        raise ArgumentError, "version is required"
+    def update_view(name, version: nil, sql_definition: nil, revert_to_version: nil, materialized: false)
```
Comment (calebthompson) [inline lib/scenic/statements.rb]: This is fine.

**Comment by jwood** [comment]
OK @calebthompson. I've added code to raise an error if both `version` and `sql_version` are specified in calls to `create_view` or `update_view`.

Sorry about the extra commits to fix the hound violations. I'll squash them if everything else looks OK.

**Comment by calebthompson** [comment]
Couple of failing tests with the new error. Can you take a look?

**Comment by calebthompson** [comment]
Interesting, the version being omitted caused the error?

**Comment by jwood** [comment]
Yes, that's the case. Because, an omitted version will default to "1". So, we end up with a version of "1" and a defined `sql_definition`...which triggers the error.

**Comment by calebthompson** [comment]
And we have the default `version: 1` in `Scenic::Statements#create_view` to allow `create_view :viewname` for the initial version.

Unfortunately I think that does make it a breaking change. We could possibly default `version` if both `version` and `sql_definition` are omitted (rather than the error we raise now), which seems slightly less bad to me.

@derekprior What do you think? The options are:

1. Leave this as-is, which will break if someone is currently doing `create_view :viewname, sql_definition: ...` because version is defaulted
2. Default `version` to 1 somewhere in the method body if neither `version` nor `sql_definition` are provided (safer, since this is an error state now)
3. Don't raise an error if both are provided, but let sql_definition override version number. Maybe omit a warning in this case as well.

My vote is for 2., as it's turning an error case into non-error (therefore not breaking) and allows both

```ruby
create_view :viewname
create_view :otherviewname, sql_definition: ...
```

**Comment by calebthompson** [comment]
(sorry to flip-flop, @jwood)

**Comment by jwood** [comment]
@calebthompson I like your suggestion of defaulting the version in the method body. Let me push up something along those lines.

**Comment by calebthompson** [comment]
LGTM. Please squash this and I'll merge.

**Comment by jwood** [comment]
@calebthompson Commits have been squashed

**Pull request merged**